filecoin/store/data.js

```diff
@@ -0,0 +1,161 @@
+import {
+  S3Client,
+  GetObjectCommand,
+  PutObjectCommand
+} from '@aws-sdk/client-s3'
+import * as CAR from '@ucanto/transport/car'
+import { sha256 } from 'multiformats/hashes/sha2'
+import { CID } from 'multiformats/cid'
+import * as raw from 'multiformats/codecs/raw'
+import { CarWriter } from '@ipld/car'
+import pRetry from 'p-retry'
+import { StoreOperationFailed, RecordNotFound } from '@web3-storage/filecoin-api/errors'
+
+/**
+ * Abstraction layer with Factory to perform operations on bucket storing
+ * data receipts.
+ *
+ * @param {string} region
+ * @param {string} bucketName
+ * @param {import('@aws-sdk/client-s3').ServiceInputTypes} [options]
+ */
+export function createDataStore(region, bucketName, options = {}) {
+  const s3client = new S3Client({
+    region,
+    ...options,
+  })
+  return useDataStore(s3client, bucketName)
+}
+
+/**
+ * @param {S3Client} s3client
+ * @param {string} bucketName
+ * @returns {import('@web3-storage/filecoin-api/storefront/api').DataStore}
+ */
+export const useDataStore = (s3client, bucketName) => {
+  return {
+    // Only used for testing storing a CAR
+    // until we hook up claims to look for data
+    put: async (bytes) => {
+      const hash = await sha256.digest(bytes)
+      const root = CID.create(1, raw.code, hash)
+
+      const { writer, out } = CarWriter.create(root)
+      writer.put({ cid: root, bytes })
+      writer.close()
+
+      const chunks = []
+      for await (const chunk of out) {
+        chunks.push(chunk)
+      }
+      const blob = new Blob(chunks)
+      const cid = await CAR.codec.link(new Uint8Array(await blob.arrayBuffer()))
+
+      const putCmd = new PutObjectCommand({
+        Bucket: bucketName,
+        Key: `${cid.toString()}/${cid.toString()}.car`,
+        Body: bytes
```

Comment:
we only need this for the tests, so that we can put something on the store. Maybe I should change the API to have an augmented store just on the tests interface, but not the actual service store. WDYT?